packages/host/app/services/store.ts

```diff
@@ -204,20 +245,75 @@ export default class StoreService extends Service {
         });
       }
     }
-    await this.handleUpdatedCard(undefined, cardOrError);
+    await this.handleUpdatedInstance(undefined, cardOrError);
     return { card, error };
   }
 
+  // This method creates a new instance in the store and return the new card ID
   async createInstance(
     doc: LooseSingleCardDocument,
     relativeTo: URL | undefined,
+    realm?: string,
   ): Promise<string | CardError> {
     await this.ready;
-    let cardOrError = await this.getCard(doc, relativeTo);
-    if (isCardInstance(cardOrError)) {
-      return cardOrError.id;
+    let token = waiter.beginAsync();
+    try {
+      await this.ready;
+      let cardOrError = await this.getInstance({
+        urlOrDoc: doc,
+        relativeTo,
+        realm,
+      });
+      if (isCardInstance(cardOrError)) {
+        return cardOrError.id;
+      }
+      return cardOrError;
+    } finally {
+      waiter.endAsync(token);
+    }
+  }
+
+  save(id: string) {
+    this.initiateAutoSaveTask.perform(id, { isImmediate: true });
+  }
+
+  // Instances that are saved via this method are eligible for garbage
+  // collection--meaning that it will be detached from the store. This means you
+  // MUST consume the instance IMMEDIATELY! it should not live in the state of
+  // the consumer.
+
+  // This method adds or creates a new instance to the store and returns an
+  // instance eligible for garbage collection.
+  async add<T extends CardDef>(
```

Comment:
this should be used in lieu of CardService.saveModel() and CardService.createFromSerialized()